app/src/main/java/com/ds/avare/views/LocationView.java

```diff
@@ -1253,41 +1257,51 @@ protected String doInBackground(Object... vals) {
             if(!mPref.useAdsbWeather()) {              
                 aireps = mService.getDBResource().getAireps(lon, lat);
                 if(isCancelled()) {
-                    return "";
+                    return null;
                 }
                 
                 wa = mService.getDBResource().getWindsAloft(lon, lat);
                 if(isCancelled()) {
-                    return "";
+                    return null;
                 }
                 
                 sua = mService.getDBResource().getSua(lon, lat);
                 if(isCancelled()) {
-                    return "";
+                    return null;
                 }
 
                 if(mLayer != null) {
                     layer = mLayer.getDate();
                 }
                 if(isCancelled()) {
-                    return "";
+                    return null;
                 }
             }
 
             navaids = mService.getDBResource().findNavaidsNearby(lat, lon);
 
+            // if we are on the navaid set this as destination
+            for (NavAid n: navaids) {
+                if (Projection.getStaticDistance(lat, lon,n.getCoords().getLatitude(), n.getCoords().getLongitude()) < 0.5) {
```

Comment:
1) is there an existing distance constant which would be better here? 
2) maybe also have priority of airport destination over VOR? i.e. check if we haven't already found an airport here